web/src/features/Dashboard/components/Dashboard.tsx

```diff
@@ -232,48 +234,46 @@ const Dashboard = (props: Props) => {
     });
   };
 
-  const getAppLicense = async ({ slug }: { slug: string }) => {
-    await fetch(`${process.env.API_ENDPOINT}/app/${slug}/license`, {
-      method: "GET",
-      headers: {
-        Authorization: Utilities.getToken(),
-        "Content-Type": "application/json",
-      },
-    })
-      .then(async (res) => {
-        const body = await res.json();
-        if (!res.ok) {
-          setState({ gettingAppLicenseErrMsg: body.error });
-          return;
-        }
-        if (body === null) {
-          setState({ appLicense: null, gettingAppLicenseErrMsg: "" });
-        } else if (body.success) {
-          setState({
-            appLicense: body.license,
-            gettingAppLicenseErrMsg: "",
-          });
-        } else if (body.error) {
-          setState({
-            appLicense: null,
-            gettingAppLicenseErrMsg: body.error,
-          });
-        }
-      })
-      .catch((err) => {
-        console.log(err);
-        setState({
-          gettingAppLicenseErrMsg: err
-            ? `Error while getting the license: ${err.message}`
-            : "Something went wrong, please try again.",
-        });
+  const {
+    data: licenseWithInterceptResponse,
+    error: licenseWithInterceptError,
+    refetch: getAppLicense,
+    isSlowLoading: isSlowLoadingLicense,
+  } = useLicenseWithIntercept();
+
+  useEffect(() => {
+    if (!licenseWithInterceptResponse) {
+      setState({ appLicense: null, gettingAppLicenseErrMsg: "" });
+      return;
+    }
+
+    if (licenseWithInterceptResponse.success) {
+      setState({
+        appLicense: licenseWithInterceptResponse.license,
+        gettingAppLicenseErrMsg: "",
       });
-  };
+      return;
+    }
+    if (licenseWithInterceptResponse.error) {
+      setState({
+        appLicense: null,
+        gettingAppLicenseErrMsg: licenseWithInterceptResponse.error,
+      });
+      return;
+    }
+    if (licenseWithInterceptError instanceof Error) {
+      setState({ gettingAppLicenseErrMsg: licenseWithInterceptError?.message });
+      return;
+    }
+
+    setState({
+      gettingAppLicenseErrMsg: "Something went wrong, please try again.",
+    });
```

Comment:
this logic replaces `getAppLicense`. Ideally we don't have a useEffect hook for this. Eventually we will factor these out and use the custom hook data directly.